csharp/src/Google.Protobuf/Reflection/MessageDescriptor.cs

```diff
@@ -104,6 +104,8 @@ internal MessageDescriptor(DescriptorProto proto, FileDescriptor file, MessageDe
             {
                 map[JsonFormatter.ToCamelCase(field.Name)] = field;
                 map[field.Name] = field;
+                if (!string.IsNullOrEmpty(field.JsonName))
```

Comment:
I can't add the comment to line 94, but could you remove that TODO at the same time?